extensions/ql-vscode/package.json

```diff
@@ -372,13 +372,23 @@
       },
       {
         "type": "object",
-        "title": "Downloading databases",
+        "title": "Adding databases",
         "order": 6,
         "properties": {
+          "codeQL.addingDatabases.allowHttp": {
+            "type": "boolean",
+            "default": false,
+            "description": "Allow databases to be downloaded via HTTP. Warning: enabling this option will allow downloading from insecure servers."
+          },
           "codeQL.databaseDownload.allowHttp": {
+            "type": "boolean",
+            "markdownDeprecationMessage": "**Deprecated**: Please use `#codeQL.addingDatabases.allowHttp#` instead.",
+            "deprecationMessage": "Deprecated: Please use codeQL.addingDatabases.allowHttp instead."
+          },
```

Comment:
As mentioned offline, users who still use this old setting name will get a deprecation message:
![image](https://github.com/github/vscode-codeql/assets/42641846/1436effc-a3e6-43bf-960c-d52ee12f72c8)

Reply:
For the record: we will no longer support the old setting name. This isn't the "formal" way of deprecating a setting, but this is a sufficiently niche/insecure setting that it shouldn't have much user impact. 

In the case where a user did have this setting enabled: If they try to download a DB from an `http` location, they'll get the error message "Must use https for downloading a database.", which should point them to check their settings.

Reply:
Discussed offline as well, but this should be fine. As said it's not the full proper way of renaming a setting, and we could aim to support both for a transition period. However we expect this setting to be almost, if not completely unused. And as Shati points out, if someone is using this the error should point them towards checking their settings and seeing the deprecation notice. This along with a changelog entry should mean that anyone who is using it sees it and updates their settings.